upstream/bootstrap.go

```diff
@@ -207,6 +213,20 @@ func (n *bootstrapper) createTLSConfig(host string) *tls.Config {
 	return tlsConfig
 }
 
+func (n *bootstrapper) loadTLSConfig(host string) *tls.Config {
```

Comment:
Why do we need a second method for that? You can simply check if `n.options.TLSClientCertificate` is specified inside `createTLSConfig`